### PR TITLE
Add a plot configuration class and `plotIndividualTimeProfile()`

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
TODO:
- [x] ~~`createPlotConfiguration()`~~ (removed after team discussion)
- [x] `ospPlotConfiguration`
- [ ] rename to something other than `ospPlotConfiguration`
- [x] rely on _tlf_ defaults; no need to choose defaults arguments for parameters ourselves
- [x] get rid of constructor parameters (rely on what R6 provides by default)
- [ ] proof of concept plot with `plotIndividualTimeProfile()`
- [ ] add tests
- [ ] add examples to manual

Related to #833
Related to #815
Part of #674

### CODE REVIEW

File: R/ospPlotConfiguration.R [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                          xlabel = NULL,
+                          ylabel = NULL,
+                          legend = NULL,
+                          xAxis = NULL,
+                          yAxis = NULL,
+                          # background = NULL,
+                          # plotArea = NULL,
+                          # panelArea = NULL,
+                          # xGrid = NULL,
+                          # yGrid = NULL,
```
Comment (IndrajeetPatil) [inline R/ospPlotConfiguration.R:95]: Not sure if we should be supporting these [parameters](https://www.open-systems-pharmacology.org/TLF-Library/reference/PlotConfiguration.html). 

image

I feel these aspects of the plot are better specified through a _tlf_ theme, which will be another parameter for all plotting functions.

Comment (PavelBal) [reply R/ospPlotConfiguration.R:95]: I would say no.

File: R/create-plot-configuration.R [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                                    errorbarsFill = NULL,
+                                    errorbarsShape = NULL,
+                                    errorbarsSize = NULL,
+                                    errorbarsLinetype = NULL,
+                                    errorbarsAlpha = NULL,
+                                    # export
+                                    plotSaveFileFormat = "png",
+                                    plotSaveFileWidth = 16,
+                                    plotSaveFileHeight = 9,
+                                    plotSaveFileDimensionUnits = "cm") {
```
Comment (IndrajeetPatil) [inline R/create-plot-configuration.R:108]: These can't be `NULL`; otherwise, the object creation fails.

**Comment by IndrajeetPatil** [comment]
Based on feedback from Pavel, here is the revised workflow:

- `createPlotConfiguration()` function creates `ospPlotConfiguration` class instance, which is nothing but a container for whatever arguments were specified by the user. This object is then more general purpose because it can be used with `base` or `ggplot`.
- Inside all plotting functions (e.g. `plotIndividualTimeProfile`), the `ospPlotConfiguration` instance is used to create an instance of `ospInternalPlotConfiguration` class via the internal function `.createOSPInternalPlotConfiguration()`.
- `ospInternalPlotConfiguration` public fields provide one-to-one mapping of fields to create an instance of the class needed for that particular plot (e.g. `TimeProfilePlotConfiguration`).

**Comment by IndrajeetPatil** [comment]
`plotIndividualTimeProfile()` draft is included only for a reference, and it will be deleted once the implementation details are accepted. It is to be part of #820.

File: R/plotIndividualTimeProfile.R [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  individualTimeProfilePlotConfiguration$background <- ospInternalPlotConfiguration$background
+
+  # Configurations for aesthetics
+  individualTimeProfilePlotConfiguration$lines <- ospInternalPlotConfiguration$lines
+  individualTimeProfilePlotConfiguration$points <- ospInternalPlotConfiguration$points
+  individualTimeProfilePlotConfiguration$ribbons <- ospInternalPlotConfiguration$ribbons
+  individualTimeProfilePlotConfiguration$errorbars <- ospInternalPlotConfiguration$errorbars
+
+  # Export configuration
+  individualTimeProfilePlotConfiguration$export <- ospInternalPlotConfiguration$export
```
Comment (IndrajeetPatil) [inline R/plotIndividualTimeProfile.R:67]: The benefit of the internal `ospInternalPlotConfiguration` class is that this code block will be **identical** across all plotting functions.

File: R/plotIndividualTimeProfile.R [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
```
Comment (IndrajeetPatil) [inline R/plotIndividualTimeProfile.R:64]: The benefit of the internal `ospInternalPlotConfiguration` class is that this code block will be **identical** across all plotting functions.

File: R/create-plot-configuration.R [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+#' @param plotSaveFileFormat,plotSaveFileWidth,plotSaveFileHeight,plotSaveFileDimensionUnits File format to which the plot needs to be saved, and the units and dimensions for saving the plot.
+#'
+#' @export
+createPlotConfiguration <- function( # labels
+                                    title = NULL,
+                                    subtitle = NULL,
+                                    xlabel = NULL,
+                                    ylabel = NULL,
+                                    # legend
+                                    legendPosition = "outsideRight",
```
Comment (msevestre) [inline R/create-plot-configuration.R:44]: legendPosition = tlf::LEgentPosition$OtusiderRight

File: R/ospPlotConfiguration.R [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+#'
+#' Note that the values this objects contains are of general-purpose utility. In
+#' other words, the public members of this class instance can be used to specify
+#' arguments for base plots, ggplot plots, or any other plotting framework.
+#'
+#' @field title,subtitle,xlabel,ylabel,legendTitle,watermark A character string
+#'   providing plot annotations for plot title, subtitle, x-axis label, y-axis
+#'   label, plot legend, watermark, respectively.
+#' @field legendPosition A character string defining the legend position
+#'   (default: `"outsideRight"`). Available options
```
Comment (Yuri05) [inline R/ospPlotConfiguration.R:16]: I would prefer `"insideTopRight"` as default

Comment (IndrajeetPatil) [reply R/ospPlotConfiguration.R:16]: As discussed yesterday, I am just going to revert all the defaults, and rely on whatever _tlf_ defaults to.

File: R/ospPlotConfiguration.R [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+#'   are:`"none"`,`"insideTop"`,`"insideTopLeft"`,
+#'   `"insideLeft"`,`"insideBottomLeft"`,`"insideBottom"`,`"insideBottomRight"`,
+#'   `"insideRight"`,`"insideTopRight"`,`"outsideTop"`,`"outsideTopLeft"`,
+#'   `"outsideLeft"`,`"outsideBottomLeft"`,`"outsideBottom"`,`"outsideBottomRight"`,
+#'    `"outsideRight"`,`"outsideTopRight"`.
```
Comment (Yuri05) [inline R/ospPlotConfiguration.R:21]: add hint that all options are accessible via `tlf::LegendPositions` e.g. `tlf::LegendPositions$insideLeft`
https://www.open-systems-pharmacology.org/TLF-Library/reference/LegendPositions.html

File: R/ospPlotConfiguration.R [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+#' @field legendCaptionColor,legendCaptionFontFamily,legendCaptionFontFace,legendCaptionAngle Aesthetic properties for the legend caption.
+#' @field xAxisTickLabels,xAxisLabelSize,xAxisLabelColor,xAxisLabelFontFamily,xAxisLabelFontFace,xAxisLabelAngle Aesthetic properties for the x-axis label.
+#' @field yAxisTickLabels,yAxisLabelSize,yAxisLabelColor,yAxisLabelFontFamily,yAxisLabelFontFace,yAxisLabelAngle Aesthetic properties for the y-axis label.
+#' @field xAxisLimits,yAxisLimits A numeric vector of axis limits for the x-and
+#'   y-axis, respectively.
+#' @field xAxisTicks,yAxisTicks A numeric vector or a function defining where to
+#'   position x-and y-axis ticks, respectively.
+#' @field xAxisScale,yAxisScale A character string defining axis scale (default:
+#'   `"lin"`). Available options are: `"lin"`, `"log"`, `"ln"`, `"discrete"`,
+#'   `"reverse"`, `"sqrt"`, `"time"`, `"date"`.
```
Comment (Yuri05) [inline R/ospPlotConfiguration.R:32]: add hint that all options are accessible via `tlf::Scaling` e.g. `tlf::Scaling$lin`
https://www.open-systems-pharmacology.org/TLF-Library/reference/LegendPositions.html
https://www.open-systems-pharmacology.org/TLF-Library/reference/Scaling.html

File: R/ospPlotConfiguration.R [hunk 9]
```diff
@@ -0,0 +1,403 @@
...
+    #' @param title,subtitle,xlabel,ylabel,legendTitle,watermark A character string
+    #'   providing plot annotations for plot title, subtitle, x-axis label, y-axis
+    #'   label, plot legend, watermark, respectively.
+    #' @param legendPosition A character string defining the legend position
+    #'   (default: `"outsideRight"`). Available options
+    #'   are:`"none"`,`"insideTop"`,`"insideTopLeft"`,
+    #'   `"insideLeft"`,`"insideBottomLeft"`,`"insideBottom"`,`"insideBottomRight"`,
+    #'   `"insideRight"`,`"insideTopRight"`,`"outsideTop"`,`"outsideTopLeft"`,
+    #'   `"outsideLeft"`,`"outsideBottomLeft"`,`"outsideBottom"`,`"outsideBottomRight"`,
+    #'    `"outsideRight"`,`"outsideTopRight"`.
```
Comment (Yuri05) [inline R/ospPlotConfiguration.R:169]: add hint that all options are accessible via `tlf::LegendPositions` e.g. `tlf::LegendPositions$insideLeft`
https://www.open-systems-pharmacology.org/TLF-Library/reference/LegendPositions.html

File: R/ospPlotConfiguration.R [hunk 10]
```diff
@@ -0,0 +1,403 @@
...
+    #' @param legendCaptionColor,legendCaptionFontFamily,legendCaptionFontFace,legendCaptionAngle Aesthetic properties for the legend caption.
+    #' @param xAxisTickLabels,xAxisLabelSize,xAxisLabelColor,xAxisLabelFontFamily,xAxisLabelFontFace,xAxisLabelAngle Aesthetic properties for the x-axis label.
+    #' @param yAxisTickLabels,yAxisLabelSize,yAxisLabelColor,yAxisLabelFontFamily,yAxisLabelFontFace,yAxisLabelAngle Aesthetic properties for the y-axis label.
+    #' @param xAxisLimits,yAxisLimits A numeric vector of axis limits for the x-and
+    #'   y-axis, respectively.
+    #' @param xAxisTicks,yAxisTicks A numeric vector or a function defining where to
+    #'   position x-and y-axis ticks, respectively.
+    #' @param xAxisScale,yAxisScale A character string defining axis scale (default:
+    #'   `"lin"`). Available options are: `"lin"`, `"log"`, `"ln"`, `"discrete"`,
+    #'   `"reverse"`, `"sqrt"`, `"time"`, `"date"`.
```
Comment (Yuri05) [inline R/ospPlotConfiguration.R:180]: add hint that all options are accessible via `tlf::Scaling` e.g. `tlf::Scaling$lin`
https://www.open-systems-pharmacology.org/TLF-Library/reference/LegendPositions.html
https://www.open-systems-pharmacology.org/TLF-Library/reference/Scaling.html

File: R/plotIndividualTimeProfile.R [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+#'   [`Theme`](https://www.open-systems-pharmacology.org/TLF-Library/reference/Theme.html)
+#'    object for `{tlf}` library.
+#' @param ospPlotConfiguration A `ospPlotConfiguration` object, which is an `R6`
+#'   class object that defines plot properties (like labels, axes scaling and
+#'   limits, legend position, etc.).
+#' @param xUnit,yUnit Units for x- and y-axes, respectively.
+#'
+#' @import tlf
+#'
+#' @export
```
Comment (Yuri05) [inline R/plotIndividualTimeProfile.R:14]: Description of the `tlfTheme `argument is missing

Comment (IndrajeetPatil) [reply R/plotIndividualTimeProfile.R:14]: You mean there should be more than this? Or you mean there should be another section for theme under the `description` tag?

```
#' @param tlfTheme A path to JSON file containing
#'   [`Theme`](https://www.open-systems-pharmacology.org/TLF-Library/reference/Theme.html)
#'    object for `{tlf}` library.
```

Comment (Yuri05) [reply R/plotIndividualTimeProfile.R:14]: I mean I need new glasses - haven't seen the tlfTheme description above 😄

**Comment by IndrajeetPatil** [comment]
Closed in favour of #871.

**Pull request closed without merging**